src/jdk.crypto.mscapi/windows/classes/sun/security/mscapi/CKeyStore.java

```diff
@@ -768,8 +781,15 @@ public void engineLoad(InputStream stream, char[] password)
 
         try {
 
-            // Load keys and/or certificate chains
-            loadKeysOrCertificateChains(getName(), getLocation(), getReadOnlyAccess());
+            if (stream != null && getRequiresKeyStoreFromFile()) {
+                // Load keys and/or certificate chains from input stream
+                byte[] blob = stream.readAllBytes();
+                loadKeysOrCertificateChainsFromMemory(blob, blob.length, getReadOnlyAccess());
+            }
+            else {
```

Comment:
what if the stream is != null but getRequiresKeyStoreFromFile() returns false?   I think there's some implicit state machine combinations here that need commenting